soh/soh/UIWidgets.cpp

```diff
@@ -765,4 +765,355 @@ namespace UIWidgets {
         float sz = ImGui::GetFrameHeight();
         return StateButtonEx(str_id, label, ImVec2(sz, sz), ImGuiButtonFlags_None);
     }
+
+    // UIWidgets V2
```

Comment:
I love this architecture of putting the tooltips and other options together in an actual object.  I would only suggest 2 things:

- First, make sure you're not losing any functionality.  For example, I recently added some really nice (not that I'm biased) functionality to the float sliders to make the +/- buttons use reasonable values no matter what the range of the slider is.  Make sure everything from the older version of the widgets gets ported to these.  
- Second, if you're going to do this, I say go whole hog and just replace them everywhere.  No need to keep the v1 versions around at all.  Totally get if you want to do that over multiple PRs though.